pep-0696.rst

```diff
@@ -345,34 +346,17 @@ subtype of one of the constraints.
    TypeVar("Ok", float, str, default=float)     # Valid
    TypeVar("Invalid", float, str, default=int)  # Invalid: expected one of float or str got int
 
-Function Defaults
-'''''''''''''''''
-
-``TypeVarLike``\ s currently can only be used where a parameter can go unsolved.
-
-.. code-block:: py
-
-   def foo(a: DefaultStrT | None = None) -> DefaultStrT: ...
-
-   reveal_type(foo(1)) # type is int
-   reveal_type(foo())  # type is str
-
-If they are used where the parameter type is known, the defaults
-should just be ignored and a type checker can emit a warning.
-
 
 Implementation
 --------------
 
 At runtime, this would involve the following changes to the ``typing``
 module.
 
-- a new class ``_DefaultMixin`` needs to be added which is a superclass
-  of ``TypeVar``, ``ParamSpec``, and ``TypeVarTuple``.
-
-   -  the type passed to default would be available as a ``__default__``
-      attribute and a sentinel would need to be added that can be
-      checked against to see if it was supplied.
+- The classes ``TypeVar``, ``ParamSpec``, and ``TypeVarTuple`` would
+  all expose the type passed to default which would be available as a
+  ``__default__`` attribute with the default being ``None`` if no
+  argument is passed and ``NoneType`` if ``default=None``.
```

Comment:
This sentence is rather long, I'd split it up. Also, should quote `default`.

And can you remind me where this `NoneType` conversion comes from?

Reply:
The NoneType conversion comes from here, converting None to NoneType is how current typing seems to handle similar cases for more see https://github.com/python/typing_extensions/pull/85.
```py
In [1]: from typing import Optional, get_args

In [2]: get_args(Optional[int])
Out[2]: (int, NoneType)
```

Reply:
Thanks, makes sense